test/provider/Makefile.am

```diff
@@ -24,24 +24,27 @@ TESTS = \
 check_PROGRAMS = rsakey eckey dhkey threadtest
 
 dhkey_SOURCES = dhkey.c
+dhkey_LDADD = -lcrypto -ldl
 if PROVIDER_FULL_LIBICA
-dhkey_LDADD = -lcrypto -lica
+dhkey_CFLAGS = -DLIBICA_NAME=\"libica.so.4\"
```

Comment:
I guess, ibmca never used an earlier libica 3 version?

Reply:
Well, this is upstream, so we assume the most current libica version. 
Don't forget, thats just a test tool for 'make check', nothing that ever gets installed on a user system.

Anyway. In configure.ac we have a vatable for the libica-version. Let me look if there is a way to pass this through into the make file....

Reply:
Fixed. Using `libicaversion` from configure.ec now. 
So the libica version can be controlled via `--with-libica-version=x` during configure.